tensorflow_recommenders_addons/dynamic_embedding/python/ops/hkv_hashtable_ops.py

```diff
@@ -0,0 +1,517 @@
+    ):
+      op = hkv_ops.tfra_hkv_hash_table_remove(self.resource_handle, keys)
+
+    return op
+
+  def clear(self, name=None):
+    """clear all keys and values in the table.
+
+    Args:
+      name: A name for the operation (optional).
+
+    Returns:
+      The created Operation.
+    """
+    with ops.name_scope(name, "%s_lookup_table_clear" % self.name,
+                        (self.resource_handle, self._default_value)):
+      op = hkv_ops.tfra_hkv_hash_table_clear(self.resource_handle,
+                                             key_dtype=self._key_dtype,
+                                             value_dtype=self._value_dtype)
+
+    return op
+
+  def prepare(self, keys, name=None, shared_name=None, is_async=False):
+    return control_flow_ops.no_op()
+
+  def lookup(self,
+             keys,
+             dynamic_default_values=None,
+             return_exists=False,
+             name=None,
+             shared_name=None):
+    """Looks up `keys` in a table, outputs the corresponding values.
+
+      The `default_value` is used for keys not present in the table.
+
+      Args:
+        keys: Keys to look up. Can be a tensor of any shape. Must match the
+          table's key_dtype.
+        dynamic_default_values: The values to use if a key is missing in the
+          table. If None (by default), the static default_value
+          `self._default_value` will be used.
+        return_exists: if True, will return a additional Tensor which indicates
+          if or not keys are existing in the table.
+        name: A name for the operation (optional).
+
+      Returns:
+        A tensor containing the values in the same shape as `keys` using the
+          table's value type.
+        exists:
+          A bool type Tensor of the same shape as `keys` which indicates
+            if keys are existing in the table.
+            Only provided if `return_exists` is True.
+
+      Raises:
+        TypeError: when `keys` do not match the table data types.
+    """
+    with ops.name_scope(
+        name,
+        "%s_lookup_table_find" % self.name,
+        (self.resource_handle, keys, self._default_value),
+    ):
+      keys = ops.convert_to_tensor(keys, dtype=self._key_dtype, name="keys")
+      with ops.colocate_with(self.resource_handle, ignore_existing=True):
+        if return_exists:
+          values, exists = hkv_ops.tfra_hkv_hash_table_find_with_exists(
+              self.resource_handle,
+              keys,
+              dynamic_default_values
+              if dynamic_default_values is not None else self._default_value,
+          )
+        else:
+          values = hkv_ops.tfra_hkv_hash_table_find(
+              self.resource_handle,
+              keys,
+              dynamic_default_values
+              if dynamic_default_values is not None else self._default_value,
+          )
+    return (values, exists) if return_exists else values
+
+  def insert(self, keys, values, name=None, shared_name=None):
```

Comment:
Could you add `scores=None` after the `values` for reserving the customized strategy supporting?